lenovo/thinkpad/x1-gen6/default.nix

```diff
@@ -0,0 +1,41 @@
+# pulled out all the x1 specific things
+# for nixos-hardware
+
+{ config, pkgs, ... }:
+
+{
+
+  boot.kernelModules = [ "acpi" "thinkpad-acpi" "acpi-call" ];
+  boot.extraModulePackages = [ config.boot.kernelPackages.acpi_call ];
+  boot.extraModprobeConfig = "options thinkpad_acpi experimental=1 fan_control=1";
+
+  environment.systemPackages = with pkgs; [
+    tlp
+  ];
+
+  # enable thermal trottle
+  services.thermald.enable = true;
+
+  # enable sound
+  sound.enable = true;
+  hardware.pulseaudio.enable = true;
```

Comment:
These probably shouldn't be here, since people may want to make their own choices about sound settings. Generally any direct settings should probably have `mkDefault` so that other people's settings will override them.